src/main/interactive_shell.cpp

```diff
@@ -104,7 +104,7 @@ InteractiveShell::InteractiveShell(ExprManager& exprManager,
   }
 
 #if HAVE_LIBREADLINE
-  if(d_in == cin) {
+  if(&d_in == &cin) {
```

Comment:
Please submit separately.
(Also please explain the motivation, it is not clear to me.)